prov/efa/src/efa_base_ep.h

```diff
@@ -39,6 +39,28 @@
 #define efa_rx_flags(efa_base_ep) ((efa_base_ep)->util_ep.rx_op_flags)
 #define efa_tx_flags(efa_base_ep) ((efa_base_ep)->util_ep.tx_op_flags)
 
+#define EFA_PROV_NAME "efa"
+#define EFA_DIRECT_PROV_NAME "efa-direct"
+
+/* TODO - CHECK THESE */
+#define EFA_EP_TYPE_IS_RDM(_info) \
+	(_info && _info->ep_attr && (_info->ep_attr->type == FI_EP_RDM) && !strcasecmp(_info->fabric_attr->name, EFA_PROV_NAME))
+
+#define EFA_EP_TYPE_IS_DIRECT(_info) \
+	(_info && _info->ep_attr && (_info->ep_attr->type == FI_EP_RDM) && !strcasecmp(_info->fabric_attr->name, EFA_DIRECT_PROV_NAME))
+
+/* TODO - CHECK THESE */
+#define EFA_EP_TYPE_IS_DGRAM(_info) \
+	(_info && _info->ep_attr && (_info->ep_attr->type == FI_EP_DGRAM))
+
+enum efa_ep_type {
+	EFA_EP_RDM = 0,
+	EFA_EP_DGRAM,
+	EFA_EP_DIRECT,
```

Comment:
efa direct shouldn't be an ep type.

Reply:
Discussed offline. Will change this enum to `efa_info_type`